src/main/scala/firrtl/transforms/CSESubAccesses.scala

```diff
@@ -42,6 +44,7 @@ object CSESubAccesses {
 
   // Replaces all right-hand side SubAccesses with References
   private def replaceOnSourceExpr(replace: SubAccess => Reference)(expr: Expression): Expression = expr match {
+    case e if flow(e) != SourceFlow => e // Stop is we ever see a non-SourceFlow
```

Comment:
nit (next pattern has different comment structure):
```suggestion
    // Stop is we ever see a non-SourceFlow
    case e if flow(e) != SourceFlow => e 
```